include/SampleFrame.h

```diff
@@ -0,0 +1,234 @@
+/*
+ * SampleFrame.h - Representation of a stereo sample
+ *
+ * Copyright (c) 2004-2009 Tobias Doerffel <tobydox/at/users.sourceforge.net>
+ * Copyright (c) 2024- Michael Gregorius
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#pragma once
+
+#ifndef SAMPLEFRAME_H
+#define SAMPLEFRAME_H
+
+#include "lmms_basics.h"
+
+#include <array>
+
+#include <cmath>
+#include <algorithm>
+
+
+namespace lmms
+{
+
+class SampleFrame
+{
+public:
+	SampleFrame() : SampleFrame(0., 0.)
+	{
+	}
+
+	SampleFrame(sample_t value) : SampleFrame(value, value)
+	{
+	}
```

Comment:
Should this really be implicit? I can see the utility in being able to fill both channels with a single value, but I think code that does so should make it clear that the conversion from a single sample to a sample frame is happening. Leaving it implicit I feel risks harming readability and introducing bugs.

Reply:
Fixed with commit 4878d1ca.